goblin/driver/cluster.py

```diff
@@ -156,13 +156,16 @@ def config_from_json(self, filename):
         """
         with open(filename, 'r') as f:
             config = json.load(f)
-        config = self._get_message_serializer(config)
+        config = self._process_config_imports(config)
         self.config.update(config)
 
-    def _get_message_serializer(self, config):
+    def _process_config_imports(self, config):
         message_serializer = config.get('message_serializer', '')
-        if message_serializer:
+        provider = config.get('provider')
+        if message_serializer and isinstance(message_serializer, str):
             config['message_serializer'] = my_import(message_serializer)
+        if provider and isinstance(provider, str):
+            config['provider'] = my_import(provider)
         return config
```

Comment:
It seems to me that this function could be simplified a bit by using the `get` methods default `None` and then just checking if its a string:
```python
    def _process_config_imports(self, config):
        message_serializer = config.get('message_serializer')
        provider = config.get('provider')
        if isinstance(message_serializer, str):
            config['message_serializer'] = my_import(message_serializer)
        if isinstance(provider, str):
            config['provider'] = my_import(provider)
        return config
```